packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -1,242 +1,270 @@
+  TokenDetectionControllerStateChangeEvent;
+
+export type AllowedEvents =
+  | NetworkControllerStateChangeEvent
+  | NetworkControllerNetworkDidChangeEvent
+  | TokenListStateChange;
+
+export type TokenDetectionControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  TokenDetectionControllerActions | AllowedActions,
+  TokenDetectionControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
 /**
- * @type TokenDetectionConfig
- *
- * TokenDetection configuration
- * @property interval - Polling interval used to fetch new token rates
- * @property selectedAddress - Vault selected address
+ * Controller that passively polls on a set interval for Tokens auto detection
+ * @property intervalId - Polling interval used to fetch new token rates
  * @property chainId - The chain ID of the current network
+ * @property selectedAddress - Vault selected address
+ * @property networkClientId - The network client ID of the current selected network
+ * @property disabled - Boolean to track if network requests are blocked
  * @property isDetectionEnabledFromPreferences - Boolean to track if detection is enabled from PreferencesController
  * @property isDetectionEnabledForNetwork - Boolean to track if detected is enabled for current network
  */
-// This interface was created before this ESLint rule was added.
-// Convert to a `type` in a future major version.
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export interface TokenDetectionConfig extends BaseConfig {
-  interval: number;
-  selectedAddress: string;
-  chainId: Hex;
-  isDetectionEnabledFromPreferences: boolean;
-  isDetectionEnabledForNetwork: boolean;
-}
-
-/**
- * Controller that passively polls on a set interval for Tokens auto detection
- */
-export class TokenDetectionController extends StaticIntervalPollingControllerV1<
-  TokenDetectionConfig,
-  BaseState
+export class TokenDetectionController extends StaticIntervalPollingController<
+  typeof controllerName,
+  TokenDetectionState,
+  TokenDetectionControllerMessenger
 > {
-  private intervalId?: ReturnType<typeof setTimeout>;
+  #intervalId?: ReturnType<typeof setTimeout>;
 
-  /**
-   * Name of this controller used during composition
-   */
-  override name = 'TokenDetectionController';
+  #chainId: Hex;
 
-  private readonly getBalancesInSingleCall: AssetsContractController['getBalancesInSingleCall'];
+  #selectedAddress: string;
 
-  private readonly addDetectedTokens: TokensController['addDetectedTokens'];
+  #networkClientId: NetworkClientId;
 
-  private readonly getTokensState: () => TokensState;
+  #disabled: boolean;
 
-  private readonly getTokenListState: () => TokenListState;
+  #isDetectionEnabledFromPreferences: boolean;
 
-  private readonly getNetworkClientById: NetworkController['getNetworkClientById'];
+  #isDetectionEnabledForNetwork: boolean;
+
+  /**
+   * Name of this controller used during composition
+   */
+
```

Comment:
Should this be removed now?

Reply:
😓 Fixed: https://github.com/MetaMask/core/pull/3609/commits/2126a22a6572811d394bc472a03edbb07e37fe21